scripts/release_archive_staging.py

```
@@ -0,0 +1,269 @@
+
+def copy_release_binaries(
+    repo: Path,
+    target: str,
+    binaries: tuple[str, ...],
+    staging_root: Path,
+) -> None:
+    """Copy Cargo release binaries into the archive staging directory."""
+    for binary in binaries:
+        source = release_binary_path(repo, target, binary)
+        if not source.is_file():
+            raise FileNotFoundError(f"release binary missing: {source}")
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)